File: library/src/main/java/com/akexorcist/roundcornerprogressbar/CenteredRoundCornerProgressBar.java
package com.akexorcist.roundcornerprogressbar;

import android.content.Context;
import android.graphics.drawable.GradientDrawable;
import android.os.Build;
import android.util.AttributeSet;
import android.widget.LinearLayout;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;

/**
 * Thank you to @redsanso for add this feature. See @see <a href="https://github.com/akexorcist/Android-RoundCornerProgressBar/issues/42">Centered Progress Bar</a>
 **/
@SuppressWarnings("unused")
@Keep
public class CenteredRoundCornerProgressBar extends RoundCornerProgressBar {

    public CenteredRoundCornerProgressBar(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CenteredRoundCornerProgressBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    public CenteredRoundCornerProgressBar(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void drawProgress(@NonNull LinearLayout layoutProgress,
                                @NonNull GradientDrawable progressDrawable,
                                float max,
                                float progress,
                                float totalWidth,
                                int radius,
                                int padding,
                                boolean isReverse) {
        super.drawProgress(layoutProgress, progressDrawable, max, progress, totalWidth, radius, padding, isReverse);
        MarginLayoutParams params = (MarginLayoutParams) layoutProgress.getLayoutParams();
        float ratio = max / progress;
        float progressWidth = (totalWidth - (padding * 2)) / ratio;
        float deltaWidth = totalWidth - progressWidth;
        params.setMargins((int) (deltaWidth / 2), params.topMargin, (int) (deltaWidth / 2), params.bottomMargin);
        layoutProgress.setLayoutParams(params);
    }
}

File: round-corner-progress-bar/src/main/java/com/akexorcist/roundcornerprogressbar/CenteredRoundCornerProgressBar.kt
package com.akexorcist.roundcornerprogressbar

import android.content.Context
import android.graphics.drawable.GradientDrawable
import android.os.Build
import android.util.AttributeSet
import android.widget.LinearLayout
import androidx.annotation.Keep
import androidx.annotation.RequiresApi

@Keep
open class CenteredRoundCornerProgressBar : RoundCornerProgressBar {
    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    override fun drawProgress(
        layoutProgress: LinearLayout,
        progressDrawable: GradientDrawable,
        max: Float,
        progress: Float,
        totalWidth: Float,
        radius: Int,
        padding: Int,
        isReverse: Boolean,
    ) {
        super.drawProgress(layoutProgress, progressDrawable, max, progress, totalWidth, radius, padding, isReverse)
        val params = layoutProgress.layoutParams as MarginLayoutParams
        val ratio = max / progress
        val progressWidth = (totalWidth - (padding * 2)) / ratio
        val deltaWidth = totalWidth - progressWidth
        params.setMargins(
            (deltaWidth / 2).toInt(),
            params.topMargin,
            (deltaWidth / 2).toInt(),
            params.bottomMargin,
        )
        layoutProgress.layoutParams = params
    }
}
